src/pages/index.js

```diff
@@ -0,0 +1,307 @@
+    if (evt.target === modal && modal.classList.contains("modal_opened")) {
+      closeModal(modal);
+    }
+  });
+});
+
+// Edit Profile
+// ---------------------------
+function handleEditFormSubmit(evt) {
+  handleSubmit(updateEditProfileForm, evt, 'Saving. . .', settings.inactiveButtonClass);
+}
+
+function updateEditProfileForm() {
+
+  return api.editUserInfo({
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value
+  })
+    .then((userInfo) => {
+
+      profileName.textContent = userInfo.name;
+      profileDescription.textContent = userInfo.about;
+      closeModal(editProfileModal);
+      // disableButton(editFormElement.querySelector(settings.submitButtonSelector), settings.inactiveButtonClass)
+
+    })
+    // .then(() => closeModal(editProfileModal));
+    .catch(console.error)
+
+}
+// -------------------------------
+
+// New Post --------------------
+function handleAddCardSubmit(evt) {
+  handleSubmit(addCardReq, evt, "Saving. . .")
+}
+
+function addCardReq() {
+  return api.addNewCard({name: cardNameInput.value, link: cardLinkInput.value})
+    .then(renderCard)
+    .then(() => closeModal(cardModal));
+}
+// ------------------------------
+
+// Avatar  - - - - - - - -
+function handleAvatarSubmit(evt) {
+  handleSubmit(addAvatarReq, evt, "Saving. . .",settings.inactiveButtonClass)
+}
+
+function addAvatarReq(){
+  return api.editUserAvatar({ avatar: avatarLinkInput.value })
+    .then((userPic) => {
+      profileImage.src = userPic.avatar;
+      closeModal(avatarModal);
+      // disableButton(editFormElement.querySelector(settings.submitButtonSelector), settings.inactiveButtonClass)
+    })
+
+}
+
+// - - - - - - - -- - - - - - - -
+
+// Delete - - - - - - - - -
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+function handleDeleteSubmit(evt) {
+  handleSubmit(addDeleteReq, evt, "Deleting. . . ");
+}
+
+function addDeleteReq() {
+  return api.deleteCard(selectedCardId)
+    .then(() =>{
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+}
+// - - - - - - - - - - - - - - - - - -
+
+
+function handleLikeCard(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-button_liked")
+  api.toggleLike(id, isLiked)
+    .then((data) => {
+      data.isLiked = !isLiked;
+      evt.target.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+//Events
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+
+  resetValidation(editFormElement, settings);
+  // resetValidation(editFormElement, [editModalNameInput, editModalDescriptionInput], settings)
+
+  openModal(editProfileModal);
+
+});
+
+
+cardModalButton.addEventListener("click", () => {
+  resetValidation(cardForm, settings)
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. 


You don't need it for the `card` or `avatar` modals